.github/workflows/build.yml

```diff
@@ -0,0 +1,145 @@
+name: "Build & Test"
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        node-version: '18'
+    - name: Get yarn cache directory path
+      id: yarn-cache-dir-path
+      working-directory: app
+      run: echo "dir=$(yarn config get cacheFolder)" >> $GITHUB_OUTPUT
+    - name: Cache Yarn Packages
+      id: yarn-cache
+      uses: actions/cache@v3
+      with:
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: yarn-cache-folder-${{ hashFiles('app/yarn.lock', 'app/.yarnrc.yml') }}
+        restore-keys: |
+          yarn-cache-folder-
+    - name: Install node dependencies
+      working-directory: app
+      run: |
+        yarn
+    - name: Lint
+      working-directory: app
+      run: |
+        yarn typecheck
+        yarn lint
+    - name: Build
+      working-directory: app
+      run: |
+        yarn build
+        tar cfJ micropad.tar.xz build/*
+    - name: Upload artifact
+      uses: actions/upload-artifact@v3
+      with:
+        name: micropad
+        path: 'app/micropad.tar.xz'
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    needs: [build]
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        node-version: '18'
+    - name: Get yarn cache directory path
+      id: yarn-cache-dir-path
+      working-directory: app
+      run: echo "dir=$(yarn config get cacheFolder)" >> $GITHUB_OUTPUT
+    - name: Cache Yarn Packages
+      id: yarn-cache
+      uses: actions/cache@v3
+      with:
+        path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+        key: yarn-cache-folder-${{ hashFiles('app/yarn.lock', 'app/.yarnrc.yml') }}
+        restore-keys: |
+          yarn-cache-folder-
+    - uses: actions/download-artifact@v3
+      with:
+        name: micropad
+    - name: Install node dependencies
+      working-directory: app
+      run: |
+        yarn
+    - name: Run Micropad in background for testing
+      run: |
+        mkdir micropad
+        tar xf micropad.tar.xz -C micropad
+        cd micropad/build
+        python3 -m http.server 3000 &
+    - name: Test
+      working-directory: app
+      run: |
+        yarn test
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    environment:
+      name: production
+      url: "https://web.getmicropad.com"
+    needs: [build, test]
+    if: github.ref == 'refs/heads/master' && github.repository == 'micropad/Micropad-Core'
+    steps:
+    - uses: actions/download-artifact@v3
+      with:
+        name: micropad
+    - run: sudo apt-get update && sudo apt-get install -y rsync
+    - run: rsync -aue "ssh -p 1276 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null" app/build/* $SSH_PATH
```

Comment:
I don't think this will work because there's no SSH key fetching in here. There's an rsync deployments action which makes this all a bit easier. I use it on one of my other projects: https://github.com/NickGeek/how-dead-inside/blob/main/.github/workflows/deploy.yml#L27

Reply:
Switched to that, you'll need to set the relevant DEPLOY secrets